asm/f64/benchAxpy_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.7
+
+package f64
+
+import (
+	"fmt"
+	"testing"
+)
+
+const (
+	testLen = 1000000
+)
+
+var (
+	a = float64(2)
+	x = make([]float64, testLen)
+	y = make([]float64, testLen)
+	z = make([]float64, testLen)
+)
+
+func init() {
+	for n := range x {
+		x[n] = float64(n)
+		y[n] = float64(n)
+	}
+}
+
+func BenchmarkAxpyUnitary(t *testing.B) {
+	naiveaxpyu := func(a float64, x, y []float64) {
+		for i, v := range x {
+			y[i] += a * v
+		}
+	}
+	tests := []struct {
+		name string
+		f    func(a float64, x, y []float64)
+	}{
+		{"AxpyUnitary", AxpyUnitary},
+		{"NaiveAxpyUnitary", naiveaxpyu},
+	}
+	for _, test := range tests {
+		for _, ln := range []uintptr{1, 3, 10, 30, 1e2, 3e2, 1e3, 3e3, 1e4, 3e4} {
```

Comment:
These only go to 3e4, but you've allocated 1e6. I'd either test harder, or allocate less (or both).